### DISCUSSION TITLE
关于封面图和首页图的一些想法

### DISCUSSION THREAD

**Discussion by symlPigeon** [theme-shoka-x/hexo-theme-shokaX#28] [Ideas]
算是我个人的一些小想法……（之前用shoka的时候感觉有些地方用着不舒服所以手动改了一些部分）
一个是设置随机封面图的时候，现在的`_image.yml`中的图片要么用完整的URL，要么是走新浪的图床（现在挂了）。能否在设置中增加一个字段，来通过相对路径来访问图片？或者说可以手动设置一个base url？大概这样子：

```yml
# in _config.shokax.yml
using_relative: true
# in _data/images.yml
- /assets/bgimgs/1.jpg
- /assets/bgimgs/2.jpg
...
```

```javascript
// script/helper/engine.js
const parseImage = function (img, size) {
    if (img.startsWith('//') || img.startsWith('http')) {
      return img
    } else if (hexo.theme.config.using_relative) { // support relative url
      return img
    }
    else {
      console.warn("sinaimg blocked all request from outside website,so don't use this format")
      return `https://tva${randomServer}.sinaimg.cn/` + size + '/' + img
    }
}
```

还有一个是能否手动选择某张图来作为顶部图片？大概像这样，可以自己选定某些图固定在顶部header部分里。

```yml
# in _config.shokax.yml
background_img:
  fixed: true
  image: /assets/index-bg.jpg
```

然后可见度监听能否设置为可选的选项？网页较多的时候这个可能会有些影响阅读……

**Comment by zkz098** [comment]
相对路径和可见度监听可选已经在写了
这个`手动选择某张图来作为顶部图片`效果是否和`experiments.gradientCover`类似，或者是其他效果

**Reply by symlPigeon** [reply]
感谢回复……gradientCover这个选项我没注意到，谢谢您提醒。话说在写的相对路径和可见度监听可选大概需要多久能够完成？

**Reply by zkz098** [reply]
0.1.4已经添加了
`experiments.gradientCover`这个功能是和gradient搭配使用的，应该不能单独启用

**Reply by symlPigeon** [reply]
emmm，如果启用`experiments.gradient`的话文章封面图都会被改成渐变色吧……感觉也许把两个功能分开会更好一些……

**Reply by zkz098** [reply]
0.1.5把`experiments.gradientCover`独立成了`experiments.fixedCover`，现在fixedCover可以独立使用了

**Comment by MoYuYuDe** [comment]
您好，请问现在能支持相对路径了吗？
